### ISSUE TITLE
Log processing fails for some runs without error

### ISSUE THREAD

**Issue by bilalshaikh42** [reproducible-biomedical-modeling/CRBM-Viz#4079]
One or more simulations failed
FAILED 61e75f35b81848012ba97b2c --- Morphogenesis checkpoint (Ciliberto et al., J Cell Biol, 2003; SBML; Fehlberg; LibSBMLSim)
FAILED 61e75f51f63eff1f7b4f1b9c --- Repressilator (Elowitz & Leibler, Nature, 2000; SBML; LSODA/LSODAR; COPASI)
FAILED 61e75f2dab69484376bc5392 --- Iron distribution (Parmar et al., BMC Syst Biol, 2017; SBML; CVODE; tellurium)
FAILED 61e75f26b81848012ba97a70 --- Heat shock protein synthesis (Szymanska et al., J Theor Biol, 2009; SBML; CVODES; AMICI)
FAILED 61e75f41ab69484376bc5455 --- Non-homologous end joining (Dolan et al., PLoS Comput Biol, 2020; BNGL; NFSim; BioNetGen)
FAILED 61e75f28b81848012ba97a83 --- Hodgkin-Huxley cell (Gleeson; NeuroML/LEMS; CVODE; Neuron)
FAILED 61e76a5cb81848012ba997b1 --- Morphogenesis checkpoint (Ciliberto et al., J Cell Biol, 2003; SBML; CVODE; tellurium)
FAILED 61e76a5bab69484376bc705a --- Morphogenesis checkpoint (Ciliberto et al., J Cell Biol, 2003; SBML; CVODE; PySCeS)
FAILED 61e76a6fab69484376bc70ed --- Repressilator (Elowitz & Leibler, Nature, 2000; SBML; CVODE; tellurium)

**Comment by bilalshaikh42** [comment]
Issue only noticed on validation

**Comment by bilalshaikh42** [comment]
This is now the bottleneck error that we are seeing.

**Comment by bilalshaikh42** [comment]
Caused by ssh connection failing. We should add retrying to the log processor to handle this case
```
[Nest] 1  - 01/19/2022, 3:41:19 PM   ERROR [SshService] Error: Timed out while waiting for handshake
62
[Nest] 1  - 01/19/2022, 3:41:26 PM   ERROR [SshService] Error: Timed out while waiting for handshake
61
[Nest] 1  - 01/19/2022, 3:41:27 PM   ERROR [SshService] Error: Timed out while waiting for handshake
60
[Nest] 1  - 01/19/2022, 3:41:27 PM   ERROR [SshService] Error: Timed out while waiting for handshake
59
[Nest] 1  - 01/19/2022, 3:41:27 PM   ERROR [SshService] Error: Timed out while waiting for handshake
58
[Nest] 1  - 01/19/2022, 3:41:27 PM   ERROR [SshService] Error: Timed out while waiting for handshake
57
[Nest] 1  - 01/19/2022, 3:41:28 PM   ERROR [SshService] Error: Connection lost before handshake
56
[Nest] 1  - 01/19/2022, 3:41:28 PM   ERROR [LogService] The job output for simulation run '61e82fec69d6393d77322acd' could not be read: Error: Connection lost before handshake
55
[Nest] 1  - 01/19/2022, 3:41:28 PM   DEBUG [AuthClientService] Using cached token for audience api.biosimulations.org for client nzDn2Y8hHWKnddeUHJ0Ntwfa60ya4bJT
54
[Nest] 1  - 01/19/2022, 3:41:28 PM     LOG [LogService] The log for simulation run '61e82fec69d6393d77322acd' was successfully saved.
53
[Nest] 1  - 01/19/2022, 3:41:28 PM   DEBUG [AuthClientService] Using cached token for audience api.biosimulations.org for client nzDn2Y8hHWKnddeUHJ0Ntwfa60ya4bJT
52
[Nest] 1  - 01/19/2022, 3:41:28 PM     LOG [SimulationStatusService] The status of simulation run '61e82fec69d6393d77322acd' was successfully updated to 'SUCCEEDED'.
51
[Nest] 1  - 01/19/2022, 3:41:28 PM     LOG [CompleteProcessor] Post-processing of simulation run '61e82fec69d6393d77322acd' completed with status 'SUCCEEDED'.
50
[Nest] 1  - 01/19/2022, 3:41:36 PM   DEBUG [Server] NatsStatus: type: "pingTimer", data: "1".
49
[Nest] 1  - 01/19/2022, 3:41:39 PM   ERROR [SshService] Error: Timed out while waiting for handshake
48
[Nest] 1  - 01/19/2022, 3:41:39 PM   ERROR [LogService] The job output for simulation run '61e82fec9b4579c3fd45944d' could not be read: Error: Timed out while waiting for handshake
47
[Nest] 1  - 01/19/2022, 3:41:39 PM   DEBUG [AuthClientService] Using cached token for audience api.biosimulations.org for client nzDn2Y8hHWKnddeUHJ0Ntwfa60ya4bJT
46
[Nest] 1  - 01/19/2022, 3:41:39 PM     LOG [LogService] The log for simulation run '61e82fec9b4579c3fd45944d' was successfully saved.
45
[Nest] 1  - 01/19/2022, 3:41:39 PM   DEBUG [AuthClientService] Using cached token for audience api.biosimulations.org for client nzDn2Y8hHWKnddeUHJ0Ntwfa60ya4bJT
44
[Nest] 1  - 01/19/2022, 3:41:39 PM     LOG [SimulationStatusService] The status of simulation run '61e82fec9b4579c3fd45944d' was successfully updated to 'SUCCEEDED'.
43
[Nest] 1  - 01/19/2022, 3:41:39 PM     LOG [CompleteProcessor] Post-processing of simulation run '61e82fec9b4579c3fd45944d' completed with status 'SUCCEEDED'.
42
[Nest] 1  - 01/19/2022, 3:41:46 PM   ERROR [SshService] Error: Timed out while waiting for handshake
41
[Nest] 1  - 01/19/2022, 3:41:46 PM   ERROR [LogService] The job output for simulation run '61e82ff0250294c03410933b' could not be read: Error: Timed out while waiting for handshake
40
[Nest] 1  - 01/19/2022, 3:41:46 PM   DEBUG [AuthClientService] Using cached token for audience api.biosimulations.org for client nzDn2Y8hHWKnddeUHJ0Ntwfa60ya4bJT
39
[Nest] 1  - 01/19/2022, 3:41:46 PM     LOG [LogService] The log for simulation run '61e82ff0250294c03410933b' was successfully saved.
38
[Nest] 1  - 01/19/2022, 3:41:46 PM   DEBUG [AuthClientService] Using cached token for audience api.biosimulations.org for client nzDn2Y8hHWKnddeUHJ0Ntwfa60ya4bJT
37
[Nest] 1  - 01/19/2022, 3:41:47 PM     LOG [SimulationStatusService] The status of simulation run '61e82ff0250294c03410933b' was successfully updated to 'SUCCEEDED'.
36
[Nest] 1  - 01/19/2022, 3:41:47 PM     LOG [CompleteProcessor] Post-processing of simulation run '61e82ff0250294c03410933b' completed with status 'SUCCEEDED'.
35
[Nest] 1  - 01/19/2022, 3:41:47 PM   ERROR [SshService] Error: Timed out while waiting for handshake
34
[Nest] 1  - 01/19/2022, 3:41:47 PM   ERROR [LogService] The job output for simulation run '61e82fec9b4579c3fd459454' could not be read: Error: Timed out while waiting for handshake
33
[Nest] 1  - 01/19/2022, 3:41:47 PM   DEBUG [AuthClientService] Using cached token for audience api.biosimulations.org for client nzDn2Y8hHWKnddeUHJ0Ntwfa60ya4bJT
32
[Nest] 1  - 01/19/2022, 3:41:47 PM     LOG [LogService] The log for simulation run '61e82fec9b4579c3fd459454' was successfully saved.
31
[Nest] 1  - 01/19/2022, 3:41:47 PM   DEBUG [AuthClientService] Using cached token for audience api.biosimulations.org for client nzDn2Y8hHWKnddeUHJ0Ntwfa60ya4bJT
30
[Nest] 1  - 01/19/2022, 3:41:47 PM   ERROR [SshService] Error: Timed out while waiting for handshake
29
[Nest] 1  - 01/19/2022, 3:41:47 PM   ERROR [LogService] The job output for simulation run '61e82fef69d6393d77322adb' could not be read: Error: Timed out while waiting for handshake
28
[Nest] 1  - 01/19/2022, 3:41:47 PM   DEBUG [AuthClientService] Using cached token for audience api.biosimulations.org for client nzDn2Y8hHWKnddeUHJ0Ntwfa60ya4bJT
27
[Nest] 1  - 01/19/2022, 3:41:47 PM     LOG [SimulationStatusService] The status of simulation run '61e82fec9b4579c3fd459454' was successfully updated to 'SUCCEEDED'.
26
[Nest] 1  - 01/19/2022, 3:41:47 PM     LOG [CompleteProcessor] Post-processing of simulation run '61e82fec9b4579c3fd459454' completed with status 'SUCCEEDED'.
25
[Nest] 1  - 01/19/2022, 3:41:47 PM     LOG [LogService] The log for simulation run '61e82fef69d6393d77322adb' was successfully saved.
24
[Nest] 1  - 01/19/2022, 3:41:47 PM   DEBUG [AuthClientService] Using cached token for audience api.biosimulations.org for client nzDn2Y8hHWKnddeUHJ0Ntwfa60ya4bJT
23
[Nest] 1  - 01/19/2022, 3:41:47 PM     LOG [SimulationStatusService] The status of simulation run '61e82fef69d6393d77322adb' was successfully updated to 'SUCCEEDED'.
22
[Nest] 1  - 01/19/2022, 3:41:47 PM     LOG [CompleteProcessor] Post-processing of simulation run '61e82fef69d6393d77322adb' completed with status 'SUCCEEDED'.
21
[Nest] 1  - 01/19/2022, 3:41:47 PM   ERROR [SshService] Error: Timed out while waiting for handshake
20
[Nest] 1  - 01/19/2022, 3:41:47 PM   ERROR [LogService] The job output for simulation run '61e82fec69d6393d77322ac7' could not be read: Error: Timed out while waiting for handshake
19
[Nest] 1  - 01/19/2022, 3:41:47 PM   DEBUG [AuthClientService] Using cached token for audience api.biosimulations.org for client nzDn2Y8hHWKnddeUHJ0Ntwfa60ya4bJT
18
[Nest] 1  - 01/19/2022, 3:41:47 PM     LOG [LogService] The log for simulation run '61e82fec69d6393d77322ac7' was successfully saved.
17
[Nest] 1  - 01/19/2022, 3:41:47 PM   DEBUG [AuthClientService] Using cached token for audience api.biosimulations.org for client nzDn2Y8hHWKnddeUHJ0Ntwfa60ya4bJT
16
[Nest] 1  - 01/19/2022, 3:41:47 PM     LOG [SimulationStatusService] The status of simulation run '61e82fec69d6393d77322ac7' was successfully updated to 'SUCCEEDED'.
15
[Nest] 1  - 01/19/2022, 3:41:47 PM     LOG [CompleteProcessor] Post-processing of simulation run '61e82fec69d6393d77322ac7' completed with status 'SUCCEEDED'.
```

**Comment by jonrkarr** [comment]
The log will have been uploaded to the S3 bucket by this point. It could be read from there instead.

**Comment by jonrkarr** [comment]
The archiveService could also be changed to get the size of the simulation output (zip file) from the S3 bucket. 

With this and the above change, we'd only have 3 SSH connections left
- Submit simulation jobs to Slurm
- Monitor simulation jobs with Slurm
- Submit Docker to Singularity conversion jobs to Slurm

**Comment by jonrkarr** [comment]
@bilalshaikh42 was this addressed by #4086?

**Comment by bilalshaikh42** [comment]
It should be, but I have not tried it under load yet

**Comment by bilalshaikh42** [comment]
In any case it should be resolved by #4093

**Comment by bilalshaikh42** [comment]
Done in #4087
